src/luaotfload-main.lua

```diff
@@ -44,25 +44,35 @@ luaotfload.fontloader_package     = "reference"    --- default: from current Con
 
 if not tex or not tex.luatexversion then
     error "this program must be run in TeX mode" --- or call tex.initialize() =)
-else
-    --- version check
-    local major    = tex.luatexversion / 100
-    local minor    = tex.luatexversion % 100
-    local revision = tex.luatexrevision --[[ : string ]]
-    local revno    = tonumber (revision)
-    local minimum  = luaotfload.min_luatex_version
-    local actual   = { major, minor, revno or 0 }
-    if actual [1] < minimum [1]
+end
+
+--- version check
+local major    = tex.luatexversion / 100
+local minor    = tex.luatexversion % 100
+local revision = tex.luatexrevision --[[ : string ]]
+local revno    = tonumber (revision)
+local minimum  = luaotfload.min_luatex_version
+local actual   = { major, minor, revno or 0 }
+if actual [1] < minimum [1]
     or actual == minimum and actual [2] < minimum [2]
     or actual == minimum and actual [2] == minimum [2] and actual [3] < minimum [3]
-    then
-        texio.write_nl ("term and log",
-                        string.format ("\tFATAL ERROR\n\z
-                                        \tLuaotfload requires a Luatex version >= %d.%d.%d.\n\z
-                                        \tPlease update your TeX distribution!\n\n",
-                                       (unpack or table.unpack) (minimum)))
-        error "version check failed"
-    end
+then
+    texio.write_nl ("term and log",
+                    string.format ("\tFATAL ERROR\n\z
+                                    \tLuaotfload requires a Luatex version >= %d.%d.%d.\n\z
+                                    \tPlease update your TeX distribution!\n\n",
+                                   (unpack or table.unpack) (minimum)))
+    error "version check failed"
+end
+
+if not utf8 then
+    texio.write_nl("term and log", string.format("\z
+        \tluaotfload: module utf8 is unavailable\n\z
+        \tutf8 is available in Lua 5.3+; engine\'s _VERSION is '%s'\n\z
+        \tThis implies that the engine is neither up-to-date nor officially supported\n\z
```

Comment:
But this isn't really the case, is it? The typical situation is LuaJITTeX: The engine isn't officaially supported, but in most cases it will be up-to-date. (Otherwise another check should fails). So maybe you could write something along the lines of "This probably means that your engine is not supported".

Reply:
Updated.